contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2130,6 +2130,63 @@ class tsqlBuilder : public tsqlCommonMutator
 		handleFloatWithoutExponent(ctx);
 	}
 
+	void exitPredicate(TSqlParser::PredicateContext *ctx) override
+	{
+		// For comparison operators directly followed by an '@@' variable, insert a space 
+		// to avoid these being parsed incorrectly by PG; this applies to the following 
+		// character sequences: =@@, >@@, <@@ as well as !=@ (i.e. a single-@ variable)
+		// Note: this issue does not occur for assignments like 'SET @v=@@spid' or column aliases like 'SELECT a=@@spid'
+
+		if ((ctx->comparison_operator()) && (ctx->expression().size() > 1))
+		{
+			std::string op = getFullText(ctx->comparison_operator());		
+			if ((op.back() == '=') || 
+			    (op.back() == '>') || 
+			    (op.back() == '<'))
+			{
+				// The operator must be followed immediately by the variable without any character in between
+				Assert(ctx->expression(1));
+				size_t startPosition = ctx->expression(1)->start->getStartIndex();
+				if ((startPosition - ctx->comparison_operator()->stop->getStopIndex()) == 1)
+				{
+					std::string var = getFullText(ctx->expression(1));
+					// The subsequent expression must be a variable starting with '@@'
+					if (var.front() == '@') 
+					{
```

Comment:
Those two if condition can be combine into one

Reply:
Indeed they can, but I am following like the principle that the second test is required only when the first succeeds, so don't do the test unless it is truly required.

Reply:
The runtime will automatically do this so won't be a problem. I suggested this since that'll help reading the code.